x-pack/plugins/enterprise_search/public/applications/app_search/components/search_ui/components/search_ui_form.tsx

```diff
@@ -0,0 +1,205 @@
+);
+const SORT_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortFieldPlaceholder',
+  { defaultMessage: 'Click to select' }
+);
+const URL_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldLabel',
+  { defaultMessage: 'URL field (Optional)' }
+);
+const URL_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldHelpText',
+  { defaultMessage: "Used as a result's link target, if applicable" }
+);
+const URL_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldPlaceholder',
+  { defaultMessage: 'Select a URL field' }
+);
+const GENERATE_PREVIEW_BUTTON_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.generatePreviewButtonLabel',
+  { defaultMessage: 'Generate a Preview' }
+);
+
+interface Option<T> extends Omit<EuiComboBoxOptionOption<T>, 'value'> {
+  value: string;
+}
+
+export const SearchUIForm: React.FC = () => {
+  const {
+    validFields,
+    validSortFields,
+    validFacetFields,
+    titleField,
+    urlField,
+    facetFields,
+    sortFields,
+  } = useValues(SearchUILogic);
+  const { engineName } = useValues(EngineLogic);
+  const {
+    onActiveFieldChange,
+    onFacetFieldsChange,
+    onSortFieldsChange,
+    onTitleFieldChange,
+    onUrlFieldChange,
+  } = useActions(SearchUILogic);
+
+  const previewHref = generatePreviwUrl(
+    {
+      fromKibana: 'true',
+      titleField,
+      urlField,
+      facets: facetFields,
+      sortFields,
+    },
+    engineName
+  );
+
+  const formatSelectOption = (fieldName: string) => {
+    return { text: fieldName, value: fieldName };
+  };
+  const formatMultiOptions = (fieldNames: string[]) =>
+    fieldNames.map((fieldName) => ({ label: fieldName, text: fieldName, value: fieldName }));
+  const formatMultiOptionsWithEmptyOption = (fieldNames: string[]) => [
+    { label: '', text: '', value: '' },
+    ...formatMultiOptions(fieldNames),
+  ];
+
+  const optionFields = formatMultiOptionsWithEmptyOption(validFields);
+  const sortOptionFields = formatMultiOptions(validSortFields);
+  const facetOptionFields = formatMultiOptions(validFacetFields);
+  const selectedTitleOption = formatSelectOption(titleField);
+  const selectedURLOption = formatSelectOption(urlField);
+  const selectedSortOptions = formatMultiOptions(sortFields);
+  const selectedFacetOptions = formatMultiOptions(facetFields);
+
+  return (
+    <div>
+      <EuiForm>
+        <EuiFormRow label={TITLE_FIELD_LABEL} helpText={TITLE_FIELD_HELP_TEXT} fullWidth>
+          <EuiSelect
+            name="title-field"
+            options={optionFields}
+            value={selectedTitleOption && selectedTitleOption.value}
+            onChange={(e) => onTitleFieldChange(e.target.value)}
+            fullWidth
+            placeholder={TITLE_FIELD_PLACEHOLDER}
```

Comment:
Just realized... are we actually using this placeholder text? AFAIK `EuiSelect` doesn't actually support the `placeholder` prop and `hasNoInitialSelection` forces an initial empty selection:

<img width="500" alt="" src="https://user-images.githubusercontent.com/549407/117196786-c87b6c00-ad9b-11eb-8c57-16247cb99247.png">

It looks like `EuiComboBox` does support `placeholder` but IMO it looks a little weird next to the EuiSelects.... WDYT about just removing the `placeholder` props & copy for simplicity?